drivers: sam0: dac: add sam0 dac support to samc21 processors

sgfeniex opened the pull request:
See previous work at #26649

The SAMC21 peripheral is a newer revision (0x201) which breaks the SYNCBUSY bits out into their own register.

This commit enables DAC support for SAMC21 parts. I have tested it using a custom board.

I would enable it for the `samc21n_xpro` board, but I don't have one to run tests on.

nandojve commented on dts/arm/atmel/samc21.dtsi:
```diff
@@ -32,6 +32,19 @@
 			prescaler = <4>;
 		};
 
+		dac0: dac@42005400 {
```
nodes are organized by address.
the devicetree change must be extract from this commit in another one.
add the dac alias.

nandojve commented on drivers/dac/dac_sam0.c:
```diff
@@ -27,6 +27,14 @@ LOG_MODULE_REGISTER(dac_sam0, CONFIG_DAC_LOG_LEVEL);
 #define SAM0_DAC_REFSEL_1 DAC_CTRLB_REFSEL_AVCC_Val
 #define SAM0_DAC_REFSEL_2 DAC_CTRLB_REFSEL_VREFP_Val
 
+/*
+ * SYNCBUSY is different between rev 0x1xx and 0x2xx
+ * The SAMC21 part has a SYNCBUSY register with individual bits
+ */
+#if defined(DAC_U2214) && (REV_DAC == 0x201)
+#define DAC_REV201
+#endif
+
```
Drop all this.

nandojve commented on drivers/dac/dac_sam0.c:
```diff
@@ -100,16 +108,26 @@ static int dac_sam0_init(const struct device *dev)
 
 	/* Reset then configure the DAC */
 	regs->CTRLA.bit.SWRST = 1;
+#if defined(DAC_REV201)
```
Change to and adjust
```diff
-#if defined(DAC_REV201)
+#if defined(DAC_STATUS_SYNCBUSY)
```

nandojve commented on drivers/dac/dac_sam0.c:
```diff
@@ -100,16 +108,26 @@ static int dac_sam0_init(const struct device *dev)
 
 	/* Enable */
 	regs->CTRLA.bit.ENABLE = 1;
+#if defined(DAC_REV201)
```
ditto

sgfeniex committed:
dts: atmel: sam0: dac: enable dac driver for SAMC21

This is only present in the SAMC21 soc

Signed-off-by: Stefan Giroux <stefan.g@feniex.com>

sgfeniex committed:
drivers: sam0: dac: add sam0 dac support to samc21 processors

The SAMC21 has a different revision of the DAC peripheral.
This change adds an ifdef to handle the changed SYNCBUSY register.

Signed-off-by: Stefan Giroux <stefan.g@feniex.com>
